mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -213,32 +221,36 @@ class StringIndexer @Since("1.4.0") (
     val labelsArray = $(stringOrderType) match {
       case StringIndexer.frequencyDesc =>
         val sortFunc = StringIndexer.getSortFunc(ascending = false)
-        countByValue(dataset, inputCols).map { counts =>
+        val orgStrings = countByValue(dataset, inputCols).toSeq
+        ThreadUtils.parmap(orgStrings, "sortingStringLabels", 8) { counts =>
```

Comment:
how is 8 picked here? should this be ~= number of driver core or something?

Reply:
Picked this number as following other places in Spark using `ThreadUtils.parmap`. I'm not sure we can use driver core config (`spark.driver.cores`) as it is not for this purpose. From the document, this config is only in cluster mode.